design/ceph/ceph-orchestrator-osd-add-remove.md

```diff
@@ -0,0 +1,266 @@
+behavior described to follow.
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+spec:
+  storage: # preexisting
+    driveGroups: # proposed new
+      - spec: < string, JSON or YAML blob to pass to ceph-volume for OSD creation >
+```
+
+Each individual `driveGroups` item's `spec` will be a simple string containing a JSON blob or YAML
+doc string for the Drive Group. `ceph-volume` feature request #1 will be used here to allow Rook to
+pass along the Drive Group JSON/YAML string without modification or inspection to `ceph-volume` when
+running the OSD provisioning job. OSD creation via Drive Groups is given priority and executed
+before any other CRD-based OSD creation methods are used (the other CRD-based creations methods are
+still used after the Drive Group method).
+
+Drive Groups should be defined only at the cluster level, not at node level. Drive Groups support a
+glob matching pattern for nodes, so every drive group should be issued on every node. Rook will not
+suspect an error for non-matching nodes due to `ceph-volume` feature 1-3.
+
+Users of `CephCluster` via Kubernetes Manifest will now be able to supply their own Drive Groups
+spec to Rook if they so desire to allow themselves greater flexibility in OSD management. Rook will
+treat this user spec as a declarative intent for the cluster.
+
+The Rook Orchestrator Mgr Module should be able to report the Drive Groups in the Rook configuration
+to the user, and the Orchestrator Module will have the ability to add new drive groups, remove
+existing drive groups, and modify existing drive groups. The user should be advised to only manage
+drive groups via one method: either via the `CephCluster` resource themselves manually or via the
+Orchestrator CLI or Dashboard interface.
+
+#### Reporting Status
+Rook should report failure messages back to the Rook Orchestrator Mgr Module via a `CephCluster`
+`status` mechanism so that the Orchestrator Module can take appropriate actions ot notify the user
+of failure. Rook may report successful statuses back if needed by the Orchestrator Module; however,
+limiting the information reported back will keep the `CephCluster` resource's `status` field smaller
+and less unwieldy for users.
+
+```yaml
+# first draft suggestions for status mechanism
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+status:
+  driveGroupOSDCreations: # proposed new
+    - node: < string, node name > # proposed new
+      driveGroups: # proposed new
+        - spec: < string, JSON or YAML blob from storage spec >
+          success: < bool, true if successful, false if not >
+          message: < string, message from results of last spec application >
+```
+
+A failure status for a Drive Group on a node should be cleared when the Drive Group is run
+successfully by `ceph-volume`.
+
+The failure status of the last-applied `ceph-volume` command with the Drive Group spec is
```

Comment:
I think that to have a static view of a procedure with several steps is not a good idea, to store only the last error message can hide the real problem.